Data_Structures_Questions.md

```diff
@@ -30,24 +30,26 @@ Answer the following questions for each of the data structures you implemented a
 
 ## Doubly Linked List
 
-1. What is the runtime complexity of `ListNode.insert_after`?
+1. What is the runtime complexity of `ListNode.insert_after`? O(1)
 
-2. What is the runtime complexity of `ListNode.insert_before`?
+2. What is the runtime complexity of `ListNode.insert_before`? O(1)
 
-3. What is the runtime complexity of `ListNode.delete`?
+3. What is the runtime complexity of `ListNode.delete`? O(1)
 
-4. What is the runtime complexity of `DoublyLinkedList.add_to_head`?
+4. What is the runtime complexity of `DoublyLinkedList.add_to_head`? O(1)
 
-5. What is the runtime complexity of `DoublyLinkedList.remove_from_head`?
+5. What is the runtime complexity of `DoublyLinkedList.remove_from_head`? O(1)
 
-6. What is the runtime complexity of `DoublyLinkedList.add_to_tail`?
+6. What is the runtime complexity of `DoublyLinkedList.add_to_tail`? O(1)
 
-7. What is the runtime complexity of `DoublyLinkedList.remove_from_tail`?
+7. What is the runtime complexity of `DoublyLinkedList.remove_from_tail`? O(1)
 
-8. What is the runtime complexity of `DoublyLinkedList.move_to_front`?
+8. What is the runtime complexity of `DoublyLinkedList.move_to_front`? O(1)
 
-9. What is the runtime complexity of `DoublyLinkedList.move_to_end`?
+9. What is the runtime complexity of `DoublyLinkedList.move_to_end`? O(1)
 
-10. What is the runtime complexity of `DoublyLinkedList.delete`?
+10. What is the runtime complexity of `DoublyLinkedList.delete`? O(1)
 
-    a. Compare the runtime of the doubly linked list's `delete` method with the worst-case runtime of the JS `Array.splice` method. Which method generally performs better?
\ No newline at end of file
+    a. Compare the runtime of the doubly linked list's `delete` method with the worst-case runtime of the JS `Array.splice` method. Which method generally performs better?
+
+> Answer: With Arrays they create gaps. Because Arrays create gaps we have to close them to do this we have loop and move through each element and close the gap. With Double Linked List you have the actual pointer the first, last, or middle elements of the list. This then allows you go straight to that pointer and remove it and pull everything forward with out having to fill up each gap through a loop.
```

Comment:
Another Answer:

The JS Array.splice method has a worst-case runtime of O(n). This is because when splicing an element out of an array, the elements that come after the element that was deleted in the array need to all be shifted up to fill the empty spot the deleted element left behind. Thus, in the worst case, we could have to shift every element forward one spot if the very first array element was spliced out of the array.